src/reanimated2/layoutReanimation/web/domUtils.ts

```diff
@@ -131,6 +132,66 @@ export function scheduleAnimationCleanup(
   setTimeout(() => removeWebAnimation(animationName), timeoutValue);
 }
 
+function addChild(child: HTMLElement, parent: Node) {
+  const childSnapshot = snapshots.get(child);
+
+  if (!childSnapshot) {
+    console.error('[Reanimated] Failed to obtain snapshot.');
+    return;
+  }
+
+  // We use that so we don't end up in infinite loop
+  child.setAttribute('data-reanimatedRemovedAfterAnimation', 'true');
+  parent.appendChild(child);
+
+  setDummyPosition(child, childSnapshot);
+
+  const originalOnAnimationEnd = child.onanimationend;
+
+  child.onanimationend = function (event: AnimationEvent) {
+    if (parent.contains(child)) {
+      parent.removeChild(child);
+    }
+
+    // Given that this function overrides onAnimationEnd, it won't be null
+    originalOnAnimationEnd?.call(this, event);
+  };
+}
+
+function findDescendantWithExitingAnimation(node: HTMLElement, root: Node) {
+  if (
+    node.hasAttribute('data-reanimatedDummy') &&
+    !node.hasAttribute('data-reanimatedRemovedAfterAnimation')
+  ) {
+    addChild(node, root);
+  }
+
+  const children = Array.from(node.children);
+
+  for (let i = 0; i < children.length; ++i) {
+    findDescendantWithExitingAnimation(children[i] as HTMLElement, root);
+  }
+}
+
+export function setObserver() {
```

Comment:
what do you think about this name?
```suggestion
export function addHtmlMutationObserver() {
```

Reply:
Looks good!